### DISCUSSION TITLE
[v5] Show an error message for unsupported links in play command

### DISCUSSION THREAD

**Discussion by DevAmiyo** [SudhanPlayz/Discord-MusicBot#885] [Ideas]
When an user passes an unsupported link is passed as a query in play command (example: https://google.com) the bot just says "🔎 Searching...".  If the link isn't supported, it should show an error message like "There was an error while searching" or "Unsupported link".

Ps. Idk if its a lavalink thing or not. Please let me know if it is...

---
Also if a facebook watch link is used, the cpu just goes upto 100% like crazy (in replit) and the bot crashes due to out of memory. Probably have something to do with erela.js-facebook idk.

**Comment by brianferri** [comment]
As far as the wrong link goes, the response was already implemented here:
![image](https://user-images.githubusercontent.com/81859084/173203916-9dcd4448-3a59-42f8-a17c-54a66ac85d67.png)

However, the actual `client.error` method was never implemented in the Bot's class. This caused the bot to miss the impending error from erela.js and/or lavalink and instead focus on the fact that `.error()` was never defined:
```bash
TypeError: client.error is not a function
    at C:\Users\main\Discord-MusicBot\commands\slash\play.js:57:14
    at async SlashCommand.run (C:\Users\main\Discord-MusicBot\commands\slash\play.js:56:15)
 ```
 Going to make a PR to fix this ASA I finish responding to this comment...
 
 Going on ahead to the actual error, erela throws this:
 ```
 ee [HeadersTimeoutError]: Headers Timeout Error
    at Timeout.ct [as _onTimeout] (C:\Users\main\Discord-MusicBot\node_modules\petitio\dist\index.js:4:5697)    at listOnTimeout (node:internal/timers:561:11)
    at processTimers (node:internal/timers:502:7) {
  code: 'UND_ERR_HEADERS_TIMEOUT'
}
```
 Which I have no clue as to what it should mean; Investigating the code just leads to some compiled mess of code which I will not look into further. It's obvious that the link is getting rejected as it's either not grabbable, doesn't contain any useful information or doesn't meet the requirements to be considered valid from the erela library, more on that in `node_modules\erela.js\dist\structures\Utils.js`.
 
 As for the facebook bug, could you please provide a link? I'd like to investigate that with the same environment you tried it out on

**Comment by DevAmiyo** [comment]
## Facebook bug
- Lets say I have a facebook watch link (https://fb.watch/dAMOVnbg1E/).
- I use the play command and pass the link as query
![play cmd](https://user-images.githubusercontent.com/84950599/173205658-25d76883-12bf-4bf4-aaf6-1ca580c7eef9.png)
- Now in replit the cpu just goes up like crazy
#### Repl Resources before:
![repl before](https://user-images.githubusercontent.com/84950599/173205716-033ce3dd-83c0-4994-9b86-45457a3c4703.png)
#### Repl Resources after using the command:
![repl after](https://user-images.githubusercontent.com/84950599/173205736-1b3c1b6b-1519-401f-9050-9a170ac831f8.png)
#### And of course the repl crashes due to out of memory:
![repl crash](https://user-images.githubusercontent.com/84950599/173205754-a0682ea6-0cfd-4db6-a5f3-9193d3887c0c.png)

**Comment by brianferri** [comment]
I've done some digging, and it seems like the whole module is just a big mess. I don't think there's anything we can do from the Bot's code to fix the facebook plugin. In all fairness I think it's also not worth the effort, It may be a cool feature to have, which would make the bot stand out even more, but until the devs on the plugin's end make the effort to fix it I suggest the plugin be removed.

**Reply by DevAmiyo** [reply]
Oh ok! I think then the plugin should be removed from the code so that it just treats it as all other invalid links and shows an error message instead of crashing the bot.

**Comment by brianferri** [comment]
> Also if a facebook watch link is used, the cpu just goes upto 100% like crazy (in replit) and the bot crashes due to out of memory. Probably have something to do with erela.js-facebook idk.

This happens with the particular link you used, all other links I've tried just crash the player, or result as `not found`
